### PR TITLE
EC2 Role

### PR DESCRIPTION
**Pull request by dpetzold**
New role for creating Ec2 profiles. It supports the following variables:

- AttachedPolicies. If defined its values are passed to ManagedPolicyArns
  in the IAM Role create.

- InstanceProfile. If true the role will be created as an instance profile.

- Path. If defined the Path will be set on the Role and the InstanceProfile.

This replaces https://github.com/cloudtools/stacker_blueprints/pull/16. The common functionality between iam_roles.Roles and iam_roles.Ec2Role was moved to RoleBaseBlueprint. Otherwise iam_roles.Roles should be the same. Please let me know what you think.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -6,4 +6,11 @@ lint:
 	flake8 stacker_blueprints
 
 test:
-	python setup.py test
+	python setup.py nosetests \
```
Comment (phobologic) [inline Makefile]: Can we pull these coverage changes into a separate pull-request?

File: stacker_blueprints/iam_roles.py [hunk 2]
```diff
@@ -73,17 +44,21 @@ def generate_policy_statements(self):
...
+    def create_policy(self, name=None):
         statements = self.generate_policy_statements()
         if not statements:
             return
 
         t = self.template
 
+        logical_name = "Policy"
+        if name:
+            logical_name += name
```
Comment (phobologic) [inline stacker_blueprints/iam_roles.py]: This reverses the name of of the policy from the blueprint before right?  Should probably make it match the way it was before so we don't have people's stacks changing out of the blue.

File: stacker_blueprints/iam_roles.py [hunk 3]
```diff
@@ -73,17 +44,21 @@ def generate_policy_statements(self):
...
+
         policy = t.add_resource(
             iam.PolicyType(
-                "{}Policy".format(name),
-                PolicyName=Sub("${AWS::StackName}-${Name}-policy", Name=name),
+                logical_name,
+                PolicyName=Sub("${AWS::StackName}-policy"),
```
Comment (phobologic) [inline stacker_blueprints/iam_roles.py]: This would result in a change as well - will this cause issues for existing users?

File: Makefile (hunk 1 above)
Comment (dpetzold) [reply Makefile]: Yup.

File: stacker_blueprints/iam_roles.py (hunk 2 above)
Comment (dpetzold) [reply stacker_blueprints/iam_roles.py]: Thanks! Didn't mean to change the order.

File: stacker_blueprints/iam_roles.py (hunk 3 above)
Comment (dpetzold) [reply stacker_blueprints/iam_roles.py]: Same as the above.

**Comment by dpetzold** [comment]
Hey @phobologic. Thanks for your feedback. I think not making breaking changes is important and that should be the role of the tests. However in this case this blueprint did not have an existing test so the breaking changes weren't caught. I would like to continue to use the shared base class to keep the blueprint DRY. Please let me know what you think.

**Pull request merged**